core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2225,4 +2225,14 @@ package object config {
       .stringConf
       .toSequence
       .createWithDefault(Nil)
+
+  private[spark] val APP_ATTEMPT_ID =
+    ConfigBuilder("spark.app.attempt.id")
```

Comment:
nit: This feels like it should be `spark.app.attemptNumber` than `spark.app.attempt.id`. Also currently this is specific to push based shuffle right? Should we add that in the documentation?

Reply:
`attemptNumber` sounds better to me.

If this conf can't be set by users, I think it's ok to keep it internal.

Reply:
AppAttemptId is currently used in lots of places to indicate the attempt id from Yarn, should we keep it as "spark.app.attempt.id" to align with all other places? Or we just want to change the configuration string to "spark.app.attemptNumber"?

Reply:
And which version should we mark it there? I added "1.3.0" since the support for multiple attempts in Yarn was added since 1.3.0. But if we change to "spark.app.attemptNumber", we should start from "3.2.0". Thoughts?

Reply:
Should we use "String" or "Integer" here for this configuration? In [SparkContext where it gets the Application AttemptID](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L340), it is a String format. But when parsing into Configuration and also resolving this in RemoteBlockPushResolver, we are using Integer to determine whether it is a new attempt.

Reply:
I see it makes sense. IMO, since it is already called `attemptId` it is fine to keep it probably like that. May be just adding slightly more info on the doc should be fine with me. Initially, I was under the impression we are generating this attemptId which is not the case.